src/site/_includes/components/Checkbox.js

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const {html} = require('common-tags');
+
+/* eslint-disable require-jsdoc,indent,max-len */
+
+/**
+ * Checkbox used to forms.
+ * @param {Object} arg Data for the checkbox.
+ */
+class Checkbox {
```

Comment:
Since you were able to style the input without needing a wrapping element I'd suggest getting rid of the shortcode entirely. The reason is because this implementation really tightly couples the label and the checkbox in a way that's not generally reusable (only works for specific layouts).